src/metatensor/models/utils/data/slice_join.py

```diff
@@ -0,0 +1,135 @@
+from typing import List
+
+import torch
+from metatensor.torch import Labels, TensorBlock, TensorMap
+
+
+def slice(tensor_map: TensorMap, index: int):
+    """
+    Slices a TensorMap along a given dimension.
+
+    :param tensor_map: The TensorMap to slice.
+    :param index: The index to select.
+
+    :return: The sliced TensorMap.
+    """
```

Comment:
Please open an issue at metatensor about these issue and link them here with a TODO that this function should be removed once they are fixed.

Reply:
Absolutely, I just realized this one wasn't needed. I've taken it off